src/easyreflectometry/sample/base_element_collection.py

```diff
@@ -48,6 +50,18 @@ def _dict_repr(self) -> dict:
         """
         return {self.name: [i._dict_repr for i in self]}
 
+    def as_dict(self, skip: Optional[List[str]] = None) -> dict:
+        """
+        Create a dictionary representation of the collection.
+
+        :return: A dictionary representation of the collection
+        """
+        this_dict = super().as_dict(skip=skip)
+        this_dict['data'] = []
+        for collection_element in self:
+            this_dict['data'].append(collection_element.as_dict(skip=skip))
```

Comment:
Maybe using `data` is too generic - how about having `element` as the key?

Reply:
Same as above